### PR TITLE
[Backport release-3_10] [Tests][Server] WMS GetPrint group and mask image

### PR DESCRIPTION
**Pull request by rldhont**
Backport a48474d from #35277

Manually backport #35937

### CODE REVIEW

File: src/server/services/wms/qgswmsrendercontext.cpp [hunk 1]
```diff
@@ -287,6 +287,19 @@ bool QgsWmsRenderContext::isValidLayer( const QString &nickname ) const
   return layer( nickname ) != nullptr;
 }
 
+QList<QgsMapLayer *> QgsWmsRenderContext::layersFromGroup( const QString &nickname ) const
+{
+  QList<QgsMapLayer *> layers;
+  if ( mLayerGroups.contains( nickname ) )
+  {
+    for ( QgsMapLayer *layer : mLayerGroups[nickname] )
```
Comment (elpaso) [inline src/server/services/wms/qgswmsrendercontext.cpp:295]: Shouldn'it is be  qgis::as_const( mLayerGroups[nickname] ) ?

File: src/server/services/wms/qgswmsrenderer.cpp [hunk 2]
```diff
@@ -589,15 +589,38 @@ namespace QgsWms
...
           {
-            QgsMapLayer *mlayer = mContext.layer( layer.mNickname );
-
-            if ( ! mlayer )
+            if ( mContext.isValidGroup( layer.mNickname ) )
             {
-              continue;
+              QList<QgsMapLayer *> layersFromGroup;
+
+              for ( QgsMapLayer *layer : mContext.layersFromGroup( layer.mNickname ) )
```
Comment (elpaso) [inline src/server/services/wms/qgswmsrenderer.cpp:596]: const iterator here too?

File: src/server/services/wms/qgswmsrendercontext.cpp (hunk 1 above)
Comment (rldhont) [reply src/server/services/wms/qgswmsrendercontext.cpp:295]: Do I have to update the method signature ?

Do I firstly make the changes on master ?

**Comment by rldhont** [comment]
The #35984 has been merged and backport here.

**Comment by rldhont** [comment]
@pblottiere can I merge this and deactivate the GetPrint tests ?

**Comment by pblottiere** [comment]
> can I merge this and deactivate the GetPrint tests ?

Honestly I'd prefer not to. I think we really have to find the culprit and fix it on master (in order to reactivate the `GetPrint` tests) before doing anything related to the `GetPrint` request. BTW I already spent some time on this particular issue but without success for now :/.

@elpaso what do you think?

**Comment by rldhont** [comment]
@pblottiere thanks for your feedback and thanks to try to isolate the segfault.

**Comment by nyalldawson** [comment]
Let's close this until the persistent default issue is fixed on master

**Pull request closed without merging**